tpm2/error.go

```diff
@@ -0,0 +1,325 @@
+package tpm2
+
+import (
+	"fmt"
+
+	"github.com/google/go-tpm/tpmutil"
+)
+
+type (
+	RcFmt0  uint8 // Format 0 error codes
+	RcFmt1  uint8 // Format 1 error codes
+	RcWarn  uint8 // Warning codes
+	RcIndex uint8 // Indexes for arguments, handles and sessions in errors
+)
+
+// Format 0 error codes.
+const (
+	RcInitialize      RcFmt0 = 0x00
+	RcFailure                = 0x01
+	RcSequence               = 0x03
+	RcPrivate                = 0x0B
+	RcHMAC                   = 0x19
+	RcDisabled               = 0x20
+	RcExclusive              = 0x21
+	RcAuthType               = 0x24
+	RcAuthMissing            = 0x25
+	RcPolicy                 = 0x26
+	RcPCR                    = 0x27
+	RcPCRChanged             = 0x28
+	RcUpgrade                = 0x2D
+	RcTooManyContexts        = 0x2E
+	RcAuthUnavailable        = 0x2F
+	RcReboot                 = 0x30
+	RcUnbalanced             = 0x31
+	RcCommandSize            = 0x42
+	RcCommandCode            = 0x43
+	RcAuthSize               = 0x44
+	RcAuthContext            = 0x45
+	RcNVRange                = 0x46
+	RcNVSize                 = 0x47
+	RcNVLocked               = 0x48
+	RcNVAuthorization        = 0x49
+	RcNVUninitialized        = 0x4A
+	RcNVSpace                = 0x4B
+	RcNVDefined              = 0x4C
+	RcBadContext             = 0x50
+	RcCPHash                 = 0x51
+	RcParent                 = 0x52
+	RcNeedsTest              = 0x53
+	RcNoResult               = 0x54
+	RcSensitive              = 0x55
+)
+
+var fmt0Msg = map[RcFmt0]string{
+	RcInitialize:      "TPM not initialized by TPM2_Startup or already initialized",
+	RcFailure:         "commands not being accepted because of a TPM failure",
+	RcSequence:        "improper use of a sequence handle",
+	RcPrivate:         "not currently used",
+	RcHMAC:            "not currently used",
+	RcDisabled:        "the command is disabled",
+	RcExclusive:       "command failed because audit sequence required exclusivity",
+	RcAuthType:        "authorization handle is not correct for command",
+	RcAuthMissing:     "5 command requires an authorization session for handle and it is not present.",
+	RcPolicy:          "policy failure in math operation or an invalid authPolicy value",
+	RcPCR:             "PCR check fail",
+	RcPCRChanged:      "PCR have changed since checked",
+	RcUpgrade:         "TPM is in field upgrade mode unless called via TPM2_FieldUpgradeData(), then it is not in field upgrade mode",
+	RcTooManyContexts: "context ID counter is at maximum",
+	RcAuthUnavailable: "authValue or authPolicy is not available for selected entity",
+	RcReboot:          "a _TPM_Init and Startup(CLEAR) is required before the TPM can resume operation",
+	RcUnbalanced:      "the protection algorithms (hash and symmetric) are not reasonably balanced. The digest size of the hash must be larger than the key size of the symmetric algorithm.",
```

Comment:
NIT: Inconsistent use of periods. All the other error strings don't use full sentences with periods.

Reply:
done